src/signals-mach.c

```diff
@@ -112,13 +112,30 @@ static kern_return_t profiler_segv_handler
                  mach_msg_type_number_t                   code_count);
 #endif
 
+#define THREAD_STATE MACHINE_THREAD_STATE
+#define THREAD_STATE_COUNT MACHINE_THREAD_STATE_COUNT
+#if defined(_CPU_X86_64_)
+typedef x86_thread_state64_t host_thread_state_t;
+typedef x86_exception_state64_t host_exception_state_t;
+#define THREAD_STATE x86_THREAD_STATE64
+#define THREAD_STATE_COUNT x86_THREAD_STATE64_COUNT
+#define HOST_EXCEPTION_STATE x86_EXCEPTION_STATE64
+#define HOST_EXCEPTION_STATE_COUNT x86_EXCEPTION_STATE64_COUNT
+
 enum x86_trap_flags {
     USER_MODE = 0x4,
     WRITE_FAULT = 0x2,
     PAGE_PRESENT = 0x1
 };
 
-static void jl_call_in_state(jl_ptls_t ptls2, x86_thread_state64_t *state,
+#elif defined(_CPU_AARCH64_)
+typedef arm_thread_state64_t host_thread_state_t;
+typedef arm_exception_state64_t host_exception_state_t;
+#define HOST_EXCEPTION_STATE ARM_EXCEPTION_STATE
```

Comment:
Note to self: This should be ARM_EXCEPTION_STATE64

Reply:
Is that a change from Apple that you are expecting? Or are you going to push a new version?

Reply:
I'll push a new version, there's still some other issues with it also.